.github/workflows/release.yml

```diff
@@ -13,51 +7,60 @@ on:
   workflow_dispatch:
     inputs:
       version:
-        description: 'release version in format 7.y.z'
+        description: 'The version that is going to be released. Should be in format 7.y.z'
         required: true
+        default: '7.y.z'
   push:
     branches: 
       - 'release'
     paths:
       - 'VERSION'
 
 jobs:
-  release-build-publish:
+  build:
+    name: Create Che Devfile Registry Release
     runs-on: ubuntu-latest
     steps:
-      - name: Clone source code
-        uses: actions/checkout@v1
+      -
+        name: "Checkout  source code"
+        uses: actions/checkout@v2
         with:
           fetch-depth: 1
           ref: release
-
-      - name: Login to Quay.io
-        uses: azure/docker-login@v1
+      - 
+        name: Prepare
+        id: prep
+        run: |
+          SHORT_SHA1=$(git rev-parse --short HEAD)
+          echo ::set-output name=short_sha1::${SHORT_SHA1}
+          VERSION=$(head -n 1 VERSION)
+          echo ::set-output name=version::${VERSION}
+          IMAGE=che-devfile-registry
+          echo ::set-output name=image::${IMAGE}
+      -
+        name: "Set up QEMU"
+        uses: docker/setup-qemu-action@v1
+      -
+        name: "Set up Docker Buildx"
+        uses: docker/setup-buildx-action@v1
+      -
+        name: "Docker quay.io Login"
+        uses: docker/login-action@v1
         with:
-          login-server: quay.io
+          registry: quay.io
           username: ${{ secrets.QUAY_USERNAME }}
           password: ${{ secrets.QUAY_PASSWORD }}
-
-      - name: Login to Docker Hub
-        uses: azure/docker-login@v1
-        with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_PASSWORD }}
-
-      - name: Build and push patched base images and happy path image
+      - 
+        name: Build and push patched base images and happy path image
         run: |
-          export TAG=$(head -n 1 VERSION)
           /bin/bash arbitrary-users-patch/happy-path/build_happy_path_image.sh --push
```

Comment:
Both the happy path image and the base images need to be built with multi-arch support as well, otherwise the devfiles inside the registry will be no-op. I would suggest modifying the build scripts and adding `PLATFORMS` files.